staging/src/k8s.io/apiserver/pkg/admission/audit_test.go

```diff
@@ -172,3 +173,43 @@ func TestWithAudit(t *testing.T) {
 		}
 	}
 }
+
+func TestWithAuditConcurrency(t *testing.T) {
+	admitAnnotations := map[string]string{
+		"plugin.example.com/foo": "foo",
+		"plugin.example.com/bar": "bar",
+		"plugin.example.com/baz": "baz",
+		"plugin.example.com/qux": "qux",
+	}
+	var handler Interface = fakeHandler{admitAnnotations: admitAnnotations, handles: true}
+	ae := &auditinternal.Event{Level: auditinternal.LevelMetadata}
+	auditHandler := WithAudit(handler, ae)
+	a := attributes()
+
+	// Simulate the scenario store.DeleteCollection
+	workers := 5
+	toProcess := make(chan int, 2*workers)
+	go func() {
```

Comment:
if we have one go routine that reads `ae.Annotations` and another go routine that writes to it via `auditHandler.Admit` - shouldn't that be enough for showing the data race? Just thinking out loud on a more simple version of such a test.

Reply:
Since my lock protection is in `auditHandler.logAnnotations`, the test requires both coroutines to read/write `ae.Annotations` via the `auditHandler.Admit` or `auditHandler.Validate`.